pycoin/tx/script/check_signature.py

```diff
@@ -97,8 +97,14 @@ def sig_blob_matches(sig_blobs, public_pairs, tmp_script, signature_for_hash_typ
         if signature_type not in sig_cache:
             sig_cache[signature_type] = signature_for_hash_type_f(signature_type, script=tmp_script)
 
-        ppp = ecdsa.possible_public_pairs_for_signature(
-            ecdsa.generator_secp256k1, sig_cache[signature_type], sig_pair)
+        try:
+            ppp = ecdsa.possible_public_pairs_for_signature(
+                ecdsa.generator_secp256k1, sig_cache[signature_type], sig_pair)
+        except ecdsa.NoSuchPointError as err:
+            from ..pay_to.ScriptType import ScriptType  # to avoid cyclical imports
+            if sig_blob != ScriptType._dummy_signature(signature_type):
```

Comment:
Not necessary, it might just make it easier to debug in some unusual situations.

I will be just as happy about it without the check. It will disregard all out of curve sigs without errors.